### ISSUE TITLE
vnstat info not showing on Ubuntu 20.04

### ISSUE THREAD

**Issue by flying-sausages** [liaralabs/swizzin_dashboard#14]
The format of the JSON has changed from mentioning `hours` to just `hour` in the version of vnstat that is in the focal repos.
This means the dashboard network status is only showing the current bandwidth, and nothing else.
![image](https://user-images.githubusercontent.com/23618693/84324116-f9452800-ab6f-11ea-8211-0a9ca7ffb33e.png)

Here is an example JSON from the new version. https://termbin.com/i479

**Comment by flying-sausages** [comment]
I tried doing this but it doesn't fix it. The ideal soultion would be to have the rights strings set based on the version of `vnstat`, so this was just to see whether this would be an approach.
```patch
diff --git a/swizzin.py b/swizzin.py
index 3bee5b2..5dbfde4 100755
--- a/swizzin.py
+++ b/swizzin.py
@@ -242,10 +242,10 @@ def vnstat(user):
     lasthour = hour - 1
     if lasthour == -1:
         lasthour = 23
-    statsh = vnstat_parse(interface, "h", "hours", hour)
-    statslh = vnstat_parse(interface, "h", "hours", lasthour)
-    statsd = vnstat_parse(interface, "d", "days", 0)
-    statsm = vnstat_parse(interface, "m", "months", 0)
+    statsh = vnstat_parse(interface, "h", "hour", hour)
+    statslh = vnstat_parse(interface, "h", "hour", lasthour)
+    statsd = vnstat_parse(interface, "d", "day", 0)
+    statsm = vnstat_parse(interface, "m", "month", 0)
     statsa = vnstat_parse(interface, "h", "total")
     #statsa = vnstat_parse(interface, "m", "total", 0)
     tops = vnstat_data(interface, "t")['interfaces'][0]['traffic']['tops']
```
Here was the error before the change above
```
Jul 06 19:52:29 exultant-mink systemd[1]: Started swizzin panel service.
Jul 06 19:53:10 exultant-mink python[12777]: [2020-07-06 19:53:10,802] ERROR in app: Exception on /stats/vnstat [GET]
Jul 06 19:53:10 exultant-mink python[12777]: Traceback (most recent call last):
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
Jul 06 19:53:10 exultant-mink python[12777]:     response = self.full_dispatch_request()
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
Jul 06 19:53:10 exultant-mink python[12777]:     rv = self.handle_user_exception(e)
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
Jul 06 19:53:10 exultant-mink python[12777]:     reraise(exc_type, exc_value, tb)
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
Jul 06 19:53:10 exultant-mink python[12777]:     raise value
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
Jul 06 19:53:10 exultant-mink python[12777]:     rv = self.dispatch_request()
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
Jul 06 19:53:10 exultant-mink python[12777]:     return self.view_functions[rule.endpoint](**req.view_args)
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/swizzin/core/htpasswd.py", line 208, in decorated
Jul 06 19:53:10 exultant-mink python[12777]:     return func(*args, **kwargs)
Jul 06 19:53:10 exultant-mink python[12777]:   File "swizzin.py", line 245, in vnstat
Jul 06 19:53:10 exultant-mink python[12777]:     statsh = vnstat_parse(interface, "h", "hours", hour)
Jul 06 19:53:10 exultant-mink python[12777]:   File "/opt/swizzin/swizzin/core/util.py", line 206, in vnstat_parse
Jul 06 19:53:10 exultant-mink python[12777]:     result = vnstat_data(interface, mode)['interfaces'][0]['traffic'][query]
Jul 06 19:53:10 exultant-mink python[12777]: KeyError: 'hours'
```

Here is the error after the change.
```
Jul 08 15:33:10 exultant-mink python[62799]: [2020-07-08 15:33:10,589] ERROR in app: Exception on /stats/vnstat [GET]
Jul 08 15:33:10 exultant-mink python[62799]: Traceback (most recent call last):
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
Jul 08 15:33:10 exultant-mink python[62799]:     response = self.full_dispatch_request()
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
Jul 08 15:33:10 exultant-mink python[62799]:     rv = self.handle_user_exception(e)
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
Jul 08 15:33:10 exultant-mink python[62799]:     reraise(exc_type, exc_value, tb)
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
Jul 08 15:33:10 exultant-mink python[62799]:     raise value
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
Jul 08 15:33:10 exultant-mink python[62799]:     rv = self.dispatch_request()
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
Jul 08 15:33:10 exultant-mink python[62799]:     return self.view_functions[rule.endpoint](**req.view_args)
Jul 08 15:33:10 exultant-mink python[62799]:   File "/opt/swizzin/swizzin/core/htpasswd.py", line 208, in decorated
Jul 08 15:33:10 exultant-mink python[62799]:     return func(*args, **kwargs)
Jul 08 15:33:10 exultant-mink python[62799]:   File "swizzin.py", line 251, in vnstat
Jul 08 15:33:10 exultant-mink python[62799]:     tops = vnstat_data(interface, "t")['interfaces'][0]['traffic']['tops']
Jul 08 15:33:10 exultant-mink python[62799]: KeyError: 'tops'
```
Not sure I missed something?

**Comment by liaralabs** [comment]
Just going to leave this here for when you have time to test.

I believe this should solve all the issues related to the new version -- however, if it doesn't, please let me know what the output is.

[vnstat.txt](https://github.com/liaralabs/swizzin_dashboard/files/4893346/vnstat.txt)

Just rename .txt to .patch and apply

**Comment by flying-sausages** [comment]
just tried it, same as above. It's probably a bit more than just the `s`
```
Jul 09 04:19:48 ssgcld python[2479525]: [2020-07-09 04:19:48,483] ERROR in app: Exception on /stats/vnstat [GET]
Jul 09 04:19:48 ssgcld python[2479525]: Traceback (most recent call last):
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
Jul 09 04:19:48 ssgcld python[2479525]:     response = self.full_dispatch_request()
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
Jul 09 04:19:48 ssgcld python[2479525]:     rv = self.handle_user_exception(e)
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
Jul 09 04:19:48 ssgcld python[2479525]:     reraise(exc_type, exc_value, tb)
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
Jul 09 04:19:48 ssgcld python[2479525]:     raise value
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
Jul 09 04:19:48 ssgcld python[2479525]:     rv = self.dispatch_request()
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
Jul 09 04:19:48 ssgcld python[2479525]:     return self.view_functions[rule.endpoint](**req.view_args)
Jul 09 04:19:48 ssgcld python[2479525]:   File "/opt/swizzin/swizzin/core/htpasswd.py", line 208, in decorated
Jul 09 04:19:48 ssgcld python[2479525]:     return func(*args, **kwargs)
Jul 09 04:19:48 ssgcld python[2479525]:   File "swizzin.py", line 251, in vnstat
Jul 09 04:19:48 ssgcld python[2479525]:     tops = vnstat_data(interface, "t")['interfaces'][0]['traffic']['tops']
Jul 09 04:19:48 ssgcld python[2479525]: KeyError: 'tops'
```

**Comment by flying-sausages** [comment]
fixed in ca71c8b
